Sources/AsyncQueue/AsyncQueue.swift

```diff
@@ -20,4 +20,70 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
-final class AsyncQueue {}
+/// A queue that enables ordered sending of events from synchronous to asynchronous code
+public final class AsyncQueue: Sendable {
+
+    // MARK: Initialization
+
+    /// Instantiates an asynchronous queue.
+    /// - Parameter priority: The priority of the tasks added to the asynchronous queue.
```

Comment:
```suggestion
    /// - Parameter priority: The baseline priority of the tasks added to the asynchronous queue.
```
I like it better if we hint that priority may go up if the system feels like it.

Reply:
Fair. If you `await` an enqueued task from a high-priority queue then yeah the priority may increase.

Apple's documentation on `Task.detached(...)` just says:
> The priority of the task.

We can do better!